mcp/src/main/java/io/modelcontextprotocol/spec/McpServerSession.java

```diff
@@ -248,7 +248,7 @@ private Mono<Void> handleIncomingNotification(McpSchema.JSONRPCNotification noti
 
 			var handler = notificationHandlers.get(notification.method());
 			if (handler == null) {
-				logger.error("No handler registered for notification method: {}", notification.method());
+				logger.warn("No handler registered for notification: {}", notification);
```

Comment:
Log abnormal behavior without indicating it's an error. 
Perhaps consider using debug level logging?